idaes/models_extra/temperature_swing_adsorption/fixed_bed_tsa0d.py

```diff
@@ -0,0 +1,4563 @@
+                        * (1 - b.bed_voidage) ** 2
+                        * b.velocity_in
+                        / b.bed_voidage**3
+                        / b.particle_dia**2
+                        / particle_sphericity**2
+                        - 1.75
+                        * (1 - b.bed_voidage)
+                        * b.velocity_in**2
+                        * b.dens_mass_gas
+                        / b.bed_voidage**3
+                        / b.particle_dia
+                        / particle_sphericity
+                    )
+                    * b.bed_height
+                )
+                * 1e-5
+            )
+
+        # add inlet port for fixed bed TSA
+        self._add_inlet_port()
+
+        # build up variables and constraints for heating step
+        self._add_heating_step()
+
+        # build up variables and constraints for cooling step
+        self._add_cooling_step()
+
+        # build up variables and constraints for pressurization step
+        self._add_pressurization_step()
+
+        # build up variables and constraints for adsorption step
+        self._add_adsorption_step()
+
+        # discretization of time domain
+        self._apply_transformation()
+
+        # add performance equations
+        self._make_performance()
+
+        # add outlet ports for fixed bed TSA
+        self._add_outlet_port()
+
+        # add emissions equations
+        self._emissions()
+
+        # add compressor
+        if self.config.compressor:
+            self._add_compressor()
+
+        # add steam calculation
+        if self.config.steam_calculation is not None:
+            self._add_steam_calc()
+
+    def _add_general_parameters(self):
+        """
+        Method to add general required parameters to run fixed bed TSA model.
+        This include fluid and wall column properties, component set,
+        etc.
+
+        """
+        self.component_list = Set(
+            initialize=["N2", "CO2", "H2O", "O2"], doc="List of components"
+        )
+        self.component_list_subset = Set(
+            initialize=["CO2", "N2"],
+            within=self.component_list,
+            doc="List of components: CO2 and N2",
+        )
+        self.component_list_subset_2 = Set(
+            initialize=["H2O", "O2"],
+            within=self.component_list,
+            doc="List of components: H2O and O2",
+        )
+        self.cp_wall = Param(
+            initialize=4.0 * 1e6,
+            units=units.J / units.m**3 / units.K,
+            doc="heat capacity of wall [J/m3/K]",
+        )
+        self.mw = Param(
+            self.component_list_subset,
+            initialize={"CO2": 0.04401, "N2": 0.02801},
+            units=units.kg / units.mol,
+            doc="Molar mass [kg/mol]",
+        )
+        self.visc_d = Param(
+            initialize=1.9e-5,
+            units=units.Pa * units.s,
+            doc="Gas phase viscosity [Pa s]",
+        )
+
+    def _add_parameters_zeolite_13x(self):
```

Comment:
Related to a previous comment, I suggest breaking these out to one or more separate files so that these can become arguments to the model. This will make it easier to add new sorbents and also help reduce the size of this file.